### PR TITLE
moving function worker service to pulsar service

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Currently, function worker service is started after the pulsar service which also starts the web service server.  This means that the web service can be serving requests for functions even though the function worker service hasn't initialized yet which will cause exceptions to be thrown.

### Modifications

Moved the worker service to be initialized and started in pulsar service

### CODE REVIEW

**Comment by jerrypeng** [comment]
@sijie @merlimat @srkukarni please review

**Comment by srkukarni** [comment]
retest this please

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 1]
```diff
@@ -390,6 +406,104 @@ public synchronized void brokerIsAFollowerNow() {
             });
 
             leaderElectionService.start();
+
+            // start function worker service if necessary
+            if (functionWorkerService.isPresent()) {
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java]: Can you move this block into a method?

Comment (jerrypeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java]: sure

**Comment by jerrypeng** [comment]
@merlimat thanks for the review.  I have addressed your comment

**Pull request merged**